jujugui/static/gui/src/app/store/env/sandbox.js

```diff
@@ -1346,15 +1353,99 @@ YUI.add('juju-env-sandbox', function(Y) {
     },
 
     /**
+      Handle Cloud.UpdateCredentials messages.
+
+      @method handleCloudUpdateCredentials
+      @param {Object} data The contents of the API arguments.
+      @param {Object} client The active ClientConnection.
+      @param {Object} state An instance of FakeBackend.
+    */
+    handleCloudUpdateCredentials: function(data, client, state) {
+      const entries = data.params.credentials;
```

Comment:
Will err if `data` is null/undefined, or doesn't have `params`, or `params` doesn't have `credentials`.